code/datums/movement_detector.dm

```diff
@@ -0,0 +1,55 @@
+/// A datum to handle the busywork of registering signals to handle in depth tracking of a movable
+/datum/movement_detector
+	var/atom/movable/tracked
+	var/datum/callback/listener
+
+/datum/movement_detector/New(atom/movable/target, datum/callback/listener)
+	if(target)
+		track(target, listener)
+
+/datum/movement_detector/Destroy()
+	untrack()
+	tracked = null
+	listener = null
+	return ..()
+
+/// Sets up tracking of the given movable atom
+/datum/movement_detector/proc/track(atom/movable/target, datum/callback/listener)
+	untrack()
+	tracked = target
+	src.listener = listener
+
+	while(ismovable(target))
```

Comment:
Is this meant to be an... if statement? This while loops seems dangerous

Reply:
Will switch it to a if and see what happens.

Reply:
works